react-native/update-font-icon.js

```diff
@@ -0,0 +1,44 @@
+// Generate our react-native icons from the fonticon stylesheet and images
+import fs from 'fs'
+import path from 'path'
+
+const css = fs.readFileSync('../desktop/renderer/fonticon.css', {encoding: 'utf8'})
+const stream = fs.createWriteStream('../shared/common-adapters/icon.paths.native.js')
+
+stream.once('open', () => {
+  stream.write('// This file is generated by update-font-icon.js, DO NOT HAND EDIT!\n')
+  stream.write('// @flow\n')
+  stream.write('\n')
+
+  stream.write('export const fontIcons = {\n')
+
+  const glyphs = []
+  let glyph = null
+  css.split('\n').forEach(line => {
+    if (glyph) {
+      const val = line.match(/content:\ \"\\(.*)\";/)[1]
+      glyphs.push(`  '${glyph}': String.fromCharCode(0x${val})`)
+      glyph = null
+    } else {
+      glyph = line.match(/^\.(.*):before/)
+      if (glyph) {
+        glyph = glyph[1]
+      }
+    }
```

Comment:
This is a little confusing, let me see if I get this right:

You iterate over every line, and try to read a glyph.
If you find a glyph, then you save that in some closed over mutable variable, so that on the next pass:
  You can match the val to that glyph.

I feel like it might be a little cleaner if you use `reduce`?
and the accumulator would be [glyphs, maybeCurrentGlyph]

I think it would look something like
```
css.split(..).reduce(([glyphs, maybeCurrentGlyph], line) => {
if (!maybeCurrentGlyph) {
  ... same as above for matching glyph
  return [glyphs, newGlyph]
} else {
  ... same as above for matching val
  glyphs.push(...)
  return [glyphs, null]
}
}, [[], null])
```
It's a little cleaner since you aren't mutating vars between iterations and the first branch of the if statement follows what you would expect the program to do (`maybeCurrentGlyph` is null)

This also might be overthinking this since this probably won't change very often